### PR TITLE
feat(menus): elgg_view_menu() now accepts an array of menu items

### PR DESCRIPTION
**Pull request by hypeJunction**
elgg_view_menu() now accepts an array of default menu items to add to the
unprepared menu bypassing menu registers. This is useful for rendering
multiple instances of the same menu on the page, e.g. filter

### CODE REVIEW

**Comment by jdalsem** [comment]
Lgtm

File: engine/lib/views.php [hunk 1]
```diff
@@ -739,6 +739,8 @@ function elgg_view_layout($layout_name, $vars = array()) {
  * @param array                      $vars An associative array of display options for the menu.
  *
  *                          Options include:
+ *                              items => an array of unprepared menu items
+ *                                       as ElggMenuItem or menu item factory options
```
Comment (mrclay) [inline engine/lib/views.php]: Can you clarify how this affects hooks?

Comment (hypeJunction) [inline engine/lib/views.php]: Hook will receive menu items injected this way. I will update the doc

Comment (mrclay) [inline engine/lib/views.php]: If folks want to avoid existing hook interference they can choose a unique menu name.

**Review by jeabakker — APPROVED** [review]

**Comment by hypeJunction** [comment]
> If folks want to avoid existing hook interference they can choose a unique menu name.

This wasn't really possible. Added a second commit to allow custom view names.

**Review by mrclay — APPROVED** [review]
Great. Any docs we should add?

**Comment by hypeJunction** [comment]
> Great. Any docs we should add?

Too lazy.

**Comment by mrclay** [comment]
> Too lazy.

hackathoners must hack!

**Pull request merged**